traitsui/testing/tester/base_classes.py

```diff
@@ -0,0 +1,44 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+from traitsui.testing.tester import locator
+
+class _IndexedEditor:
+    """ Wrapper for Editor and index """
+
+    target_class = None
+    handlers = []
+    solvers = []
+
+    def __init__(self, target, index):
+        self.target = target
+        self.index = index
```

Comment:
I agree this refactoring is worth doing, and is a neat idea.

This is referring to the pattern where we choose not to resolve the given locator (e.g. `locator.Index`) until later, instead, we want to keep the target and the locator through.

This would be a pattern useful for things like TableEditor, where we have to keep the location information (e.g. row and column indices for a cell) and editor through until when we perform a mouse click or a key click etc.

This is a pattern different from things like RangeEditor, where we can resolve the textbox widget and throw the location information away.

So I am wondering, what do you think about further abstracting the `index` here to just `location`. Maybe we can rename `target` to `ancestor` or `source` (something like that).  Then this object is not at all "index" specific. We will need to add the `locator_class` as class attribute for subclass to set, but that's about it.

(Related: It might be good to rename `target` anyway so we won't be writing `wrapper.target.target`)